apache/superset
# feat: Implement currencies formatter for saved metrics

### SUMMARY
WARNING: This PR might be considered a breaking change. Currently the d3formats defined in dataset editor for saved metrics worked only in Table and Pivot Table charts. Now they will also work in viz types listed below and they take priority over d3 formats defined in control panel. That means that charts that use metrics that have custom formats defined on dataset level might change their displayed number formats.

Enables users to set a currency formatter for saved metrics in Datasource edit modal.

Currently supported currencies are: `"USD", "EUR", "GBP", "INR", "MXN", "JPY", "CNY"` (+ user can manually type other currency codes in the currency picker). More currency codes can be added in `config.py` by modifying array `CURRENCIES`.

Currently supported charts are: Table, Pivot Table, Big Number (+ with trendline), (Time)Series Echarts, Pie, Gauge, Funnel, Treemap.

The number is formatted using the d3 format specified either for saved metric in datasource editor (takes priority) or in control panel. If user picks a d3 formatter that adds `%` or currency signs, the special signs are ignored.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


https://github.com/apache/superset/assets/15073128/281ec0b5-3506-4dbb-a709-50080fa9906b


### TESTING INSTRUCTIONS
0. If you're running this PR locally, run the migration first
1. Open 1 of the supported charts
2. Open datasource editor, add/edit a saved metric and set a currency
3. Use the metric with currency in the chart
4. Verify that values are formatted with currency sign

### ADDITIONAL INFORMATION


  Current: 0.13 s
  10+: 0.12 s
  100+: 0.12 s
  1000+: 0.13 s

CC @yousoph
